File: Aeds 2/Tp 1/Palindromo/c/Recursivo/palindromoRec.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h> 
// clear && gcc palindromoRec.c && ./a.out < pub.in > result.txt
bool isPalindromo(int left,int right,char word[]){

    if (left>=right){
        return true;
    }
    
    if (word[left] < 128 || word[right] < 128) {
        if (word[left] != word[right]){
            return false;
        }
    }

    return isPalindromo(left + 1, right - 1, word);
}


int main(){
    char word[500];
    
    scanf(" %[^\r\n]",word);
    
    int tam =  strlen(word);
    int i  = 0;
    int j = tam/2;
    
    while(strcmp(word,"FIM")){
        
        tam = strlen(word);
        if(isPalindromo(0, tam - 1, word)==true){
            printf("SIM\n");
        }
        else
            printf("NAO\n");

        scanf(" %[^\r\n]",word);
    }


    return 0;
}

File: Aeds_2/Tp 1/Palindromo/java/Recursivo/palindromoRec.java
import java.util.*;

 class palindromoRec {
    public static Scanner scanner = new Scanner(System.in);
    
    public static Boolean isPalindromo(String word,int tam,int i){
        Boolean ehpalindromo = true;

        if(i>tam){
            return ehpalindromo;
        }
       else if(word.charAt(i) != word.charAt(tam)){
            return ehpalindromo = false;
        }
        else
            return isPalindromo(word, tam-1, i+1);
        
            

    }

    public static void main(String[] args){
        String word = scanner.nextLine();
        int tam = word.length();
        int i = 0;
        while(!word.equals("FIM")){
            if(isPalindromo(word,tam-1,i)==true){
                System.out.println("SIM");
            }
            else
            System.out.println("NAO");
           
            word =  scanner.nextLine();
            tam = word.length();
            i = 0;

        }
    }
}
